app/grandchallenge/core/validators.py

```diff
@@ -188,3 +190,72 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not (self == other)
+
+
+@deconstructible
+class ViewContentValidator:
+    """Validates view_content JSON against governing rules."""
+
+    def __call__(self, value):
```

Comment:
I think a custom validator is not the way to go here. It spreads the validation logic related to the view_content over multiple places (validator and the model's clean method), making it harder to maintain, I think. You now have a dangling `check_consistent_viewports` check that this validator does not (and cannot) do, but which you want to check as well when validating your example. 

I think it would be better to leave all validation (except the JSON schema, of course) on the model level. You could add a `clean_view_content` method to the `HangingProtocolMixin` that calls all the necessary sub-clean-functions, and then call that method in the `ViewContentExampleMixin` with `self.instance.clean_view_content()`

Reply:
Making sure to have set the example on the instance before, of course `self.instance.view_content = example_view_content`

Reply:
I did it as suggested, like
```
self.instance.view_content = example_view_content
self.instance.clean_view_content()
```

However, we are setting the example as the value of the actual field `view_content`. In update forms that is already overwriting the original `view_content` value (as I can see in the debugger). Fortunately, the JSON widget is not using that instance value, and therefore, the original value is correctly kept in the field on the rendered form.
However, it does not feel ideal/neat to validate this way. What do you think?

Reply:
Ah yes, that is a good point. I'm actually not sure what the best way is. Maybe @jmsmkn can comment?

Reply:
Why not provide the view content as an argument to the validation method rather than acting on an attribute?

Reply:
Thanks! I provided the view content as an argument to the validation method.
Done.